lib/organize_letters.rb

```diff
@@ -0,0 +1,36 @@
+require_relative './ds/heap'
+
+def data = Struct.new(:value, :frequency)
+
+def organize_letters(string)
+  letters = Hash.new(0)
+  queue = Heap.new
+  string_size = string.size
+  new_string = ""
+  
+  string.each_char { |char| letters[char] += 1 }
+
+  letters.each do |letter, frequency|
+    queue.insert_smallest_to_highest(data.new(letter, frequency))
+  end
```

Comment:
for the example, we will have the queue like this:

![image](https://github.com/ImMPrada/short-ds-challenges/assets/26731448/35b6dba7-bc6d-404d-b062-1ec886205783)